contracts/core/price-aggregator/tests/price_aggregator_blackbox_test.rs

```diff
@@ -148,27 +158,31 @@ impl PriceAggregatorTestState {
         price: u64,
         err_message: &str,
     ) {
-        self.world.sc_call(
-            ScCallStep::new()
-                .from(from)
-                .call(self.price_aggregator_contract.submit(
-                    EGLD_TICKER,
-                    USD_TICKER,
-                    submission_timestamp,
-                    price,
-                    DECIMALS,
-                ))
-                .expect(TxExpect::user_error("str:".to_string() + err_message)),
-        );
+        self.world
+            .tx()
+            .from(&from.to_address())
+            .to(PRICE_AGGREGATOR)
+            .typed(price_aggregator_proxy::PriceAggregatorProxy)
+            .submit(
+                EGLD_TICKER,
+                USD_TICKER,
+                submission_timestamp,
+                price,
+                DECIMALS,
+            )
+            .with_result(ExpectStatus(4))
+            .with_result(ExpectMessage(err_message))
+            .run();
     }
 
     fn vote_slash_member(&mut self, from: &AddressValue, member_to_slash: Address) {
-        self.world.sc_call(
-            ScCallStep::new().from(from).call(
-                self.price_aggregator_contract
-                    .vote_slash_member(member_to_slash),
-            ),
-        );
+        self.world
+            .tx()
+            .from(&from.to_address())
```

Comment:
Should work directly as .from(from)

If it doesn't, then we need some more conversions.